tests/src/core/testqgsproject.cpp

```diff
@@ -959,6 +962,145 @@ void TestQgsProject::testEmbeddedGroupWithJoins()
   QCOMPARE( vl->fields().count(), 5 );
 }
 
+void TestQgsProject::testAsynchronousLayerLoading()
+{
+  std::unique_ptr<QgsProject> project = std::make_unique<QgsProject>();
+
+  QStringList meshFilters;
+  meshFilters << QStringLiteral( "*.nc" ) << QStringLiteral( "*.2dm" );
+  QStringList rasterFilters;
+  rasterFilters << QStringLiteral( "*.asc" ) << QStringLiteral( "*.tif" );
+  QStringList vectorFilters;
+  vectorFilters << QStringLiteral( "*.shp" );
+
+  QStringList meshFiles;
+  meshFiles << QStringLiteral( "lines.2dm" )
+            << QStringLiteral( "mesh_shading.nc" )
+            << QStringLiteral( "mesh_z_ws_d.nc" )
+            << QStringLiteral( "mesh_z_ws_d_vel.nc" )
+            << QStringLiteral( "quad_and_triangle.2dm" )
+            << QStringLiteral( "quad_flower.2dm" )
+            << QStringLiteral( "quad_flower_to_edit.2dm" )
+            << QStringLiteral( "quad_flower_to_edit_expected.2dm" )
+            << QStringLiteral( "refined_quad_flower.2dm" )
+            << QStringLiteral( "simplebox_clm.nc" )
+            << QStringLiteral( "trap_steady_05_3D.nc" )
+            << QStringLiteral( "with_flat_face.2dm" )
+            << QStringLiteral( "with_free_vertex_in_mesh.2dm" )
+            << QStringLiteral( "with_manifold_face.2dm" )
+            << QStringLiteral( "with_unique_shared_vertex.2dm" );
```

Comment:
is it worth testing also with mesh layers if mdal is not enabled for parallel loading?